<launch>
	<group ns="camera">
			<node name="image_proc" pkg="image_proc" type="image_proc"/>
		</group>

		<node name="camera" pkg="avt_vimba_camera" type="mono_camera_node" output="screen">
			<param name="guid" value="DEV_000F310285CA"/>
			<param name="ip_address" value="169.254.100.1"/>
			<param name="camera_info_url" value="file://$(find avt_vimba_camera)/calibrations/calibration_50-0503343290.yaml"/>
			<param name="frame_id" value="left_optical"/>
			<param name="trig_timestamp_topic" value=""/>
			<param name="show_debug_prints" value="true"/>

			<!-- Trigger mode:
				1. Freerun
				2. SyncIn1
				3. SyncIn2
				4. SyncIn3
				5. SyncIn4
				6. FixedRate
				7. Softwaremono_camera_node: /home/mag2/tutorial_ws/src/avt_vimba_camera/src/avt_vimba_camera.cpp:123: void avt_vimba_camera::AvtVimbaCamera::start(std::string, std::string, bool): Assertion `cam_guid_str == guid_str' failed.

			-->
			<param name="trigger_source" value="FixedRate"/>

			<!-- Acquisition mode:
				1. Continuous
				2. SingleFrame
				3. MultiFrame
				4. Recorder
			-->
			<param name="acquisition_mode" value="Continuous"/>

			<!-- Acquisition rate in fps -->
			<param name="acquisition_rate" value="30"/>

			<!-- Pixel format:
				1. Mono8
				2. Mono12
				3. Mono12Packed
				4. BayerRG8
				5. BayerRG12Packed
				6. BayerGR12
				7. RGB8Packed
				8. BGR8Packed
			-->
			<param name="pixel_format" value="BayerRG8"/>
			<!-- Exposure in us -->
			<param name="exposure" value="20000"/>
			<!-- Gain in dB -->
			<param name="gain" value="500"/>
			<param name="balance_ratio_abs" value="0.5"/>

			<!-- Auto control
				1. Off
				2. Once
				3. Continuous
			-->
			<param name="exposure_auto" value="Off"/>
			<param name="gain_auto" value="Off"/>
			<param name="whitebalance_auto" value="Off"/>

			<param name="binning_x" value="1"/>
			<param name="binning_y" value="1"/>
			<param name="decimation_x" value="1"/>
			<param name="decimation_y" value="1"/>
			<param name="x_offset" value="8"/>
			<param name="y_offset" value="9"/>
			<param name="width" value="800"/>
			<param name="height" value="720"/>

			<param name="stream_bytes_per_second" value="45000000"/>
		</node>
  
  <!--
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/camera/image_color"/>
    <param name="autosize" value="true" />
  </node>

  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />  
  -->
  <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" />  
  <node name="tracker" pkg="mag_tracking" type="tracker_node" output="screen">
    <remap from="image" to="/camera/image_raw"/>
  </node>
 
</launch>
